src/openms/source/FORMAT/MRMFeatureQCFile.cpp

```diff
@@ -28,265 +28,180 @@
 // ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 //
 // --------------------------------------------------------------------------
-// $Maintainer: Douglas McCloskey $
-// $Authors: Douglas McCloskey $
+// $Maintainer: Douglas McCloskey, Pasquale Domenico Colaianni $
+// $Authors: Douglas McCloskey, Pasquale Domenico Colaianni $
 // --------------------------------------------------------------------------
 
 #include <OpenMS/FORMAT/CsvFile.h>
 #include <OpenMS/FORMAT/MRMFeatureQCFile.h>
 #include <OpenMS/ANALYSIS/OPENSWATH/MRMFeatureQC.h>
 #include <OpenMS/SYSTEM/File.h>
 #include <OpenMS/DATASTRUCTURES/StringListUtils.h>
+#include <boost/regex.hpp>
 
 namespace OpenMS
 {
-
-  MRMFeatureQCFile::MRMFeatureQCFile()
+  void MRMFeatureQCFile::load(const String& filename, MRMFeatureQC& mrmfqc) const
   {
-  }
-
-  MRMFeatureQCFile::~MRMFeatureQCFile()
-  {
-  }
-
-  void MRMFeatureQCFile::load(const String & filename,
-    MRMFeatureQC & mrmfqc)
-  {
-    // read in the .csv file
-    char is = ',';
-    bool ie = false; 
-    Int first_n = -1;
-    CsvFile::load(filename, is, ie, first_n);
-
-    // parse the file
-    std::map<String,int> headers;
-    std::map<String,int> params_headers;
-    StringList line, header;
-    for (size_t i = 0; i < CsvFile::rowCount(); ++i)
+    CsvFile csv(filename, ',', false, -1);
+    StringList sl;
+    std::map<String, Size> headers;
+    if (csv.rowCount() > 0) // avoid accessing a row in an empty file
+    {
+      csv.getRow(0, sl);
+    }
+    for (Size i = 0; i < sl.size(); ++i)
+    {
+      headers[sl[i]] = i; // for each header found, assign an index value to it
+    }
+    if (headers.count("component_name")) // load component file
```

Comment:
This check will result in the wrong behavior if a component group QC file has a column for "component_name".  This could easily happen if someone is just copying and pasting values over from one spread sheet to another.